snuba/subscriptions/executor.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from concurrent.futures import Future, ThreadPoolExecutor
+
+from snuba import settings
+from snuba.api.query import ClickhouseQueryResult, parse_and_run_query
+from snuba.datasets.dataset import Dataset
+from snuba.request import Request
+from snuba.subscriptions.consumer import Tick
+from snuba.subscriptions.data import Subscription
+from snuba.subscriptions.scheduler import ScheduledTask
+from snuba.utils.metrics.timer import Timer
+
+
+executor_pool = ThreadPoolExecutor(max_workers=settings.SUBSCRIPTIONS_MAX_CONCURRENT_QUERIES)
```

Comment:
Let's make this an instance attribute on `Executor` and add a `close` method to `Executor` that cleanly shuts down the executor. (I'm not sure this is totally necessary, but seems like a good idea.)

Reply:
Do we want this as an instance variable though? At the moment, this will limit the total number of concurrent queries for the entire consumer, but if we make it per `Executor` then the limit only applies to each Executor instance... not sure whether we'll have more than one.

Reply:
I think it's reasonable to assume for now that there's only one `Executor` (note: maybe we should rename this to `SubscriptionExecutor` or `SubscriptionQueryExecutor` to avoid confusing with `ThreadPoolExecutor`) since by convention we run one consumer per dataset.

If we need to make sure that the total number of query execution threads started by the Python process are bounded, i'd rather do that by explicitly passing a `ThreadPoolExecutor` to the constructor and initializing it in the CLI where the consumer, worker, etc. are initialized (same idea as what we were talking about yesterday with the subscription store being passed the Redis client instead of depending on the module global.)